### ISSUE TITLE
gcc: error: unrecognized command-line option '-Qunused-arguments'

### ISSUE THREAD

**Issue by adamjstewart** [jax-ml/jax#25488]
### Description

When building jaxlib with GCC (yes, I know this is experimental) I see the following build error:
```
gcc: error: unrecognized command-line option '-Qunused-arguments'
```
It seems like the build is using clang-specific flags even when `--use_clang=false`.

Looks like these flags were added in #23787

### System info (python version, jaxlib version, accelerator, etc.)

* Python: 3.13.0
* Jaxlib: 0.4.37
* GCC: 13.2.0

[Build log](https://github.com/user-attachments/files/18132977/spack-build-out.txt)

**Comment by ybaturina** [comment]
Hi @adamjstewart ,
Yes, `build.py` script allows building wheels only with `clang`. This change was intentional.
However, you can try building it via `bazel`. Please try the following options for your `bazel build` command:
`--config=cuda --repo_env=CC=<GCC compiler path> --@local_config_cuda//:cuda_compiler=nvcc --host_crosstool_top="@local_config_cuda//crosstool:toolchain" --crosstool_top="@local_config_cuda//crosstool:toolchain" --extra_toolchains="@local_config_cuda//crosstool:toolchain-linux-x86_64"`

**Comment by adamjstewart** [comment]
Is there any documentation for this if (once) it changes again? I'm unfamiliar with bazel and anything involving bazel always feels daunting. Also, I need a cross-platform solution for a package manager. How does that last arg vary across macos/linux and x86_64/aarch64/ppc64le?

**Comment by ybaturina** [comment]
Hi @adamjstewart , I've made some changes in build.py. Would you try this PR please and let me know if it works for you?
https://github.com/jax-ml/jax/pull/25531

**Comment by adamjstewart** [comment]
Thanks, I tried your PR and it does fix the `-Qunused-arguments` issue, but I still have an unknown build failure:

* [build log](https://github.com/user-attachments/files/18174910/spack-build-out.txt)

It isn't even clear to me what the error is, any idea how to debug this?

**Comment by ybaturina** [comment]
Hi Adam, in the provided log I see the error below:
`gcc: fatal error: Killed signal terminated program cc1plus`
I looked it up in the web, seems like it might be OOM issue (see [here](https://brainly.com/question/43022741) for example).

**Comment by adamjstewart** [comment]
Ah that would make sense. Is there a recommended way to control the number of parallel jobs used to build jaxlib?

**Comment by ybaturina** [comment]
Please try this:
```
python build/build.py build --wheels=jaxlib --use_clang=false --bazel_options=--jobs=20
```

**Comment by adamjstewart** [comment]
That seems to work! I've now succeeded in building JAX on x86_64. I'm still seeing aarch64 build issues, but I can open another issue for that. This issue can be closed once #25531 is merged.

**Comment by ybaturina** [comment]
hi @adamjstewart the PR https://github.com/jax-ml/jax/pull/25531 is merged.
